src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -21,11 +23,29 @@ public UserService(final UserRepository userRepository) {
         return userRepository.findAll();
     }
 
-    public void addUser(User user) {
-        userRepository.save(user);
+    public User saveUser(User user) {
+        return userRepository.save(user);
     }
 
     public User getUserById(long id) throws UserNotFoundException {
         return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
     }
+
+    public User banUser(UserDto userDtoBanned) throws UserNotFoundException {
+        User user = userRepository.findById(userDtoBanned.getId()).orElseThrow(UserNotFoundException::new);
+
+        user.setStatus("banned");
+
+        return userRepository.save(user);
+    }
+
+    public User generateKey(UserDto userDto) throws UserNotFoundException {
+        User user = userRepository.findById(userDto.getId()).orElseThrow(UserNotFoundException::new);
+
+        long key = new Random().nextLong() + user.getId();
```

Comment:
Nie jestem pewien co chciałeś uzyskać dodając na końcu id. Identyfikację użytkownika? W ten sposób to nie zadziała, ponieważ zsumujesz dwie wartości i nie będzie wiadomo co zostało dodane jako "id". Zastanawiam się też czy w kluczu nie powinna być zawarta informacja pozwalająca zweryfikować ważność? Na przykład:
<4 losowe cyfry>DDMMRRHHmmiD
wtedy wiemy, że cyfry 5-14 pozwolą określić moment utworzenia i co za tym idzie sprawdzić aktualność klucza
a cyfry od 15 do ostatniej identyfikują użytkownika

Reply:
Nic specjalnego szczerze mówiąc, zostawiłem na ten moment to co miałem wcześniej. Myślałem wlaśnie o czymś takim co piszesz ale dalej w sumie nie wiem co mam zrobić z tym kluczem. Czy podstawiać go pod userkey użytkownika czy może coś innego z nim zrobić.

Reply:
Dwie sprawy
- Random jesli faktycznie jest potrzebny, powinien byc stalą. 
`private static final RADOM = new Random`
- do czego ma słuzyc ten endpoint i ta metoda ?

Reply:
Co do pierwszego, poprawione, za chwilę wrzucę commit,  co do drugiego "wygenerowanie losowego klucza ważnego godzinę po podaniu poprawnych danych użytkownika" i właśnie nie wiem jak to rozwiązać, czy to podstawiać w pole userKey czy w jakiś inny sposób.